en/docs/design/api-security/api-authentication/secure-apis-using-api-keys.md

```diff
@@ -166,6 +166,56 @@ api_key_keystore = "APIKeyKeyStore"
 api_key_alias = "<custom-alias>"
 ```
 
+### Configuring Certificate Rotation
+
+In WSO2 API Manager, API keys are signed by the Key Manager (KM) using a private key. The Gateway validates these tokens using the corresponding public certificate.
+
+So, when rotating certificates, you must:
+- Generate a new key pair.
+- Update configs in both Key Manager and Gateway.
+- Ensure old tokens are still validated during the transition (grace period).
+
+In order to rotate the certificate,
+1. Generate a new key pair in the same keystore or a new one:
+
+    ```
+    keytool -genkeypair -alias wso2carbon_new -keyalg RSA -keysize 2048 -validity 3650 -keystore wso2carbon.jks -storepass wso2carbon -keypass wso2carbon
```

Comment:
We have this primary secondary keystores as well. We need to give steps on those too.